### PR TITLE
4066 vuetify datatable refactor

### PR DESCRIPTION
**Pull request by pavelpostnikov1**
externalPagination, externalSorting and crudActions refactored

### CODE REVIEW

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/tables/DataTable.java [hunk 1]
```diff
@@ -279,37 +254,6 @@ public boolean elIsRed(int colNum, int elNum) {
...
-    public void createWithoutSave(String name, String cal, String fat, String carbs, String prot) {
-        UIElement newItem = find("//span[contains(text(), 'New Item')]");
-        UIElement cancel = $("//div[@role='document']//span[contains(text(), 'Cancel')]");
-        newItem.click();
-        create(name, cal, fat, carbs, prot);
-        cancel.click();
-    }
-
     @JDIAction("Change required {name} element name")
     public void editElement(int elNum, String newEl) {
```
Comment (pnatashap) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/tables/DataTable.java:258]: это тоже общий функционал?

**Review by pnatashap — CHANGES_REQUESTED** [review]

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/DataTablesTests.java [hunk 2]
```diff
@@ -168,34 +173,38 @@ public static void slotsTableTest() {
...
     public static void cRUDActionsTableTest() {
-        cRUDActionsTable.createWithSave("Egg", "72", "4.8", "0.4", "6.3");
+        newItemButton.click();
+        newItemCard.fill("Egg", "72", "4.8", "0.4", "6.3");
+        newItemCard.saveButton().click();
         cRUDActionsTable.assertThat().elementName(1, "Egg");
 
-        //cRUDActionsTable.createWithoutSave("Milk", "61", "3.3", "4.8", "3.2");
-        //jdiAssert(cRUDActionsTable.getColumn(1).get("Milk").isExist(), Matchers.is(false));
+        newItemButton.click();
```
[the remark this answers is not in the document]
Comment (pavelhedge) [reply jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/DataTablesTests.java:209]: done

**Comment by pavelpostnikov1** [comment]
Searched through DataTable and DataTableAssert classes. Moved to test classes everything that cannot be implemented using bare vuetify datatables. Had to change DataIterator class too because it extends DataTable and some if that functionality was used there. Added method getColorName for Chip and corresponding assert to ChipAssert to check colors in Item table.

**Review by vklonin — CHANGES_REQUESTED** [review]
Please consider following changes in your code:
- [ ] before each test make sure that a table you a testing is shown on a screen (e.g. customFilter.show())
- [ ] add a slot removal step in a slotTableTest after each slot assertion (e.g.  slotsSelect.close())
- [ ] change locator for all elements based on a mdi icon - seems Vuetify devs moved on form mdi
- [ ] delete elementsValueInColumn and elementsInColumn and change them to method of BaseTable -> count()
- [ ] refactor collapseGroup and expandGroup based on a assumption - to not use icons (currently Vuetify is not using icons specified in our methods (e.g.mdi-minus)), however if you are unable to find a way to work in this case without icons then rise a bug to change Vuetify test page to adjust tables test page on a test site.
- [ ] refactor a group method in order to get possibility to group by column name not by just first found word 'group' (eg 
Screen Shot 2022-07-26 at 5 20 20 PM


- [ ] consider to find a better way to work with cells in table than firstColumnHasElement, secondColumnHasElement, firstColumnElement, secondColumnElement - what if we will have much more columns? I mean it is possible to just use getCell() of BaseTable
- [ ] rename elementSelected and elementNotSelected to cellIsSelected and so on - it is better to understand what are we checking exactly.
- [ ] change itemTable to a DataTable and check colors according to logic on a Vuetify page
   (   getColor (calories) {
        if (calories > 400) return 'red'
        else if (calories > 200) return 'orange'
        else return 'green'  )
- [ ] refactor an expand() and isExpanded methods of DataTable - shouldn't be based on exact "7" column, but on a class v-data-table__expand-icon.
- [ ] refactor ExternalPaginationTable: 
    - itemsPerPage please base input anchored to "Items per page" - it might looks fragile however anchoring to just some input field might be causing a confusion - you also can find a better anchor than I suggested.
    - refactor firstPage() as a method which accepts a page number - if there will be like 3 of 9 pages 
- [ ] sortWithButtonAsc/sortWithButtonDsc refactor to one method with toggle behavior like it is meant by button text - it will be better for jdi user to work seamlessly with ui then.

rename filterableTableSearhField

**Review by vklonin — APPROVED** [review]

**Review by vklonin — APPROVED** [review]

**Pull request merged**